### ISSUE TITLE
PDF degradation during printing

### ISSUE THREAD

**Issue by AdrienGeoorge** [ONLYOFFICE/DocumentServer#1844]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
OnlyOffice degrades the PDF format by handling fonts badly

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
When you click on print in OnlyOffice, the Adobe Acrobat Reader software opens. Printing in Acrobat is not possible. The message "Cannot print" appears.
After analysis it seems that this is due to the font ("Type3 unknown")

**What is the expected behavior?**
Printing should work

**Did this work in previous versions of DocumentServer?**
No. Tested with versions 6.4.2 / 7.0.1 / 7.1.1

**DocumentServer version:**
7.1.1

**Operating System:**
Windows / Linux

**Browser version:**
Firefox 91.6.1esr

**Comment by ShockwaveNN** [comment]
Hi, please describe your issue in more details, maybe record a video because

> When you click on print in OnlyOffice, the Adobe Acrobat Reader software opens

Why is Adobe Reader opens on print? On print function we just generate PDF file and save that, we do not force opening of Adobe Reader

And since it's a font problem - maybe some specific fonts installed on your setup that trigger that error

**Comment by AdrienGeoorge** [comment]
Hello,
Sorry for the late reply.

When I click on the "print" button my RTF file is converted to PDF, then downloaded locally to my computer.
It is then opened in Adobe Acrobat Reader because it is the configuration of my Firefox.

When I click on the print button in Adobe Acrobat Reader I get an error message: "Unable to print the document" and then "Error while printing the document".

You will find a video attached.

https://user-images.githubusercontent.com/36088368/184144869-d388dfca-1ad1-49a7-9215-16b6333f0d05.mp4

**Comment by ShockwaveNN** [comment]
@AdrienGeoorge If Acrobat reader opens this file fine and cannot print - I don't think it's an issue in our application, could you try some other PDF reader software instead of Acrobat and see if the issue persists?

**Comment by AdrienGeoorge** [comment]
@ShockwaveNN We have tested printing from Firefox, Chrome, Acrobat Reader and PDF X-change.
The result is the same: printing is impossible.

**Comment by ShockwaveNN** [comment]
Could you attach your original file and pdf file you've trying to print?

**Comment by AdrienGeoorge** [comment]
You will find our basic .rtf file and the .pdf file that is generated by OnlyOffice when you click on the print button.

[Files.zip](https://github.com/ONLYOFFICE/DocumentServer/files/9309184/Files.zip)

**Comment by LuckySpecialist** [comment]
Hi @AdrienGeoorge!
Firstly, we are considering issues that concern only our application.
If you try to print a document from our application and get an error, most likely the problem is on our side and we will try to figure out the reason.
If you use a third-party application and get an error there, it is likely that it is no longer in our power to fix it. And yet there are also two options here: either we wrote the file incorrectly or the application processes it incorrectly.

Secondly, you have described two completely different scenarios and want them to work the same way.
I will explain:
You are using the installed ONLYOFFICE distribution on your remote server, which has a limited set of fonts and may not have the ones you need. Therefore, our application generates a file based on the set of fonts that it has. The fact that the font has the right name in the application interface does not mean that they are available to you at the moment.
And a locally installed product with all the necessary fonts on the other hand, which has all of them at its disposal when writing a file.

And thirdly, we are only considering the problems of the current release (version 7.1.1 at the moment). Unless you are a paid customer and at the same time use an old version of the application for some reason.
The files that you provided were created by an earlier version of our application.

Anyway, I checked your steps on my test environment and couldn't repeat the errors using Acrobat Reader v22.002.20191 (64-bit). Besides, I didn't quite understand what kind of printer you use for printing. In my case it was MS Print to PDF.

Total:
Please consider the above and perhaps you will determine the problem yourself.
If you are still sure that we have it, then provide more complete and up-to-date information so that we can repeat it ourselves.
Thank you for your understanding and feedback.

**Comment by AdrienGeoorge** [comment]
Hello!

We have added all the fonts we use on our server that hosts ONLYOFFICE.

The printing problem we are experiencing is still present with ONLYOFFICE version 7.1.1 and the latest version of Adobe Acrobat available to date.

We also tried to print with XPS Printer and the problem is the same.

https://user-images.githubusercontent.com/36088368/184827678-1d0709e8-7efb-41eb-812c-a820bd6bd44d.mp4

**Comment by AdrienGeoorge** [comment]
@LuckySpecialist It is when I click on the print button in OnlyOffice that my document opens in Adobe Acrobat afterwards. So it is your software that generates the PDF...

**Comment by LuckySpecialist** [comment]
Hi, @AdrienGeoorge!
As long as you write to me what I don't argue with and ignore the other questions that I asked, we won't solve anything.
The file is generated by our application - a fact.
The file is printed by a third-party application, in which you get an error - a fact, but for some reason you don't want to accept it.
If I follow your logic, then I can write like this:
On my server, I click the Print button and a browser window appears with a printer selection. I select a printer from the list and send the document to print. Printing is successful. There is no problem.

I do not know what and how you have configured on your server, whether you have tried to reproduce the problem on another PC, etc. 
In other words, if you do not cooperate in finding the causes of your problem, I will not be able to help.

**Comment by letroisieme** [comment]
> Secondly, you have described two completely different scenarios and want them to work the same way. I will explain: You are using the installed ONLYOFFICE distribution on your remote server, which has a limited set of fonts and may not have the ones you need. Therefore, our application generates a file based on the set of fonts that it has. The fact that the font has the right name in the application interface does not mean that they are available to you at the moment. And a locally installed product with all the necessary fonts on the other hand, which has all of them at its disposal when writing a file.

I am encountering an issue similar to what you describe, as in my local desktop editor app has the font I'm using (which is installed in windows) but the file is on my onlyoffice personal workspace without the same font, and I fail to export the file I'm working on in desktop editor with the right font

I opened an issue here https://github.com/ONLYOFFICE/DesktopEditors/issues/1172

How can I make it so it's possible to export a pdf with that font I'm trying to use?

Thanks for any input

**Comment by capybara614** [comment]
Hi,

I am using the docker version of DocumentServer, version 7.3.2.8, in which I have installed some custom fonts by putting them in /usr/share/fonts inside the container and running the documentserver-generate-allfonts.sh script.

When I try to print a pdf from a document in Onlyoffice that uses these custom fonts, the file is generated successfully but text written in one of the fonts shows up as garbled characters. 

To help reproduce the issue, I am joining a [zip archive](https://github.com/ONLYOFFICE/DocumentServer/files/10872762/Onlyoffice.Font.Rendering.Issue.zip) with an sample document in docx format, 2 pdf generated by Onlyoffice, one from the browser and using the ConvertService API as I'm unsure if that has an effect along with a control which was generated from LibreOffice Writer where the fonts render correctly. I have also included the fonts inside the archive.

Hope that helps, cheers.
